backends/dpdk/examples/ipsec.p4

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2023 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <core.p4>
+#include <dpdk/pna.p4>
+
+
+#define inbound(meta) (istd.direction == PNA_Direction_t.NET_TO_HOST)
+#define outbound(meta) (istd.direction == PNA_Direction_t.HOST_TO_NET)
+
+//
+// Packet headers.
+//
+header ethernet_t {
+	bit<48> dst_addr;
+	bit<48> src_addr;
+	bit<16> ether_type;
+}
+
+#define ETHERNET_ETHERTYPE_IPV4 0x0800
+
+header ipv4_t {
+	bit<4>  version;
+	bit<4>  ihl;
+	bit<6>  dscp;
+	bit<2>  ecn;
+	bit<16> total_length;
+	bit<16> identification;
+	bit<3>  flags;
+	bit<13> fragment_offset;
+	bit<8>  ttl;
+	bit<8>  protocol;
+	bit<16> header_checksum;
+	bit<32> src_addr;
+	bit<32> dst_addr;
+}
+
+#define IP_PROTOCOL_ESP 0x32
+
+header esp_t {
+	bit<32> spi;
+	bit<32> seq;
+}
+
+struct headers_t {
+	ethernet_t ethernet;
+	ipv4_t ipv4;
+	esp_t esp;
+}
+
+//
+// Packet metadata.
+//
+struct metadata_t {
+	bit<32> next_hop_id;
+        bool bypass;
+}
+
+//
+// Extern objects.
+//
+ipsec_accelerator() ipsec;
+
+//
+// Parser.
+//
+parser MainParserImpl(
+	packet_in pkt,
+	out   headers_t hdrs,
+	inout metadata_t meta,
+	in    pna_main_parser_input_metadata_t istd)
+{
+	bool from_ipsec;
+	ipsec_status status;
+
+	state start {
+		from_ipsec = ipsec.from_ipsec(status);
```

Comment:
Can you show how 'status' might be used in the parser?

Reply:
It can be tested in "if" or "verify" statements. 
The from_ipsec method can well be called outside the parser and the status can tested in if.